### PR TITLE
added instance for prod

### LINKED ISSUE #13043: prod

### PR DESCRIPTION
**Pull request by AkshayC2001**
Close #13043

### CODE REVIEW

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -230,6 +230,12 @@ def diagonal(self, *, offset=0, axis1=0, axis2=1):
             axis2=axis2,
         )
 
+    def compress(self, value, *, axis=None):
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:233]: This should be removed

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @AkshayC2001 

The unit tests are failing try checking if your data generation is correct😊
You need to generate castable dtypes for backends kindly do checkout this example to have an overview https://github.com/unifyai/ivy/blob/3cef1bba6b7abb4fd9d66d7933bff442aa87c9c1/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py#L77

Thanks

**Comment by AkshayC2001** [comment]
> Hey @AkshayC2001 
> 
> The unit tests are failing try checking if your data generation is correct😊
> You need to generate castable dtypes for backends kindly do checkout this example to have an overview https://github.com/unifyai/ivy/blob/3cef1bba6b7abb4fd9d66d7933bff442aa87c9c1/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py#L77
> 
> Thanks

Maybe because I have added compress by mistake. I will try removing it

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @AkshayC2001 

The tests are failing due to that you're not generating `keepdims` you need to add it in test example generation.
After that try to run tests locally from your end and fix any error that will come up
Thanks😊

**Comment by AkshayC2001** [comment]
ivy_tests/test_ivy/test_frontends/test_jax/test_jax_lax_operators.py this test is giving error. so this function is working fine i guess. can you confirm this

**Comment by AkshayC2001** [comment]
> Hey @AkshayC2001
> 
> The tests are failing due to that you're not generating `keepdims` you need to add it in test example generation. After that try to run tests locally from your end and fix any error that will come up Thanks😊

I will do. Thank you

**Comment by AkshayC2001** [comment]
hello @hirwa-nshuti 
are all tests supposed to pass except lint/check-formatting?

**Comment by AkshayC2001** [comment]
hello @hirwa-nshuti,
do i have added both keepdims and intials generations but still tests are not passing.

**Comment by AkshayC2001** [comment]
Hello @hirwa-nshuti 
Tests were passing locally but it is failing here. Can you check the code

File: ivy_tests/test_ivy/test_frontends/test_numpy/helpers.py [hunk 2]
```diff
@@ -176,7 +176,7 @@ def test_frontend_function(*args, where=None, **kwargs):
 # noinspection PyShadowingNames
 def handle_where_and_array_bools(where, input_dtype, test_flags):
     if isinstance(where, list) or isinstance(where, tuple):
-        where = where[0]
+        where = ivy.array(where[0], dtype='bool')
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_numpy/helpers.py:179]: This one should be reverted

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @AkshayC2001 

The problem is mainly with the way that you're generating tests examples which is resulting into overflow. Kindly refer to the tests of the numpy fronted function here https://github.com/unifyai/ivy/blob/59e6f8fd57978501724dcb6425709171e1368584/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py#L77

And do check the comments below.
Thanks

**Comment by AkshayC2001** [comment]
> Hey @AkshayC2001
> 
> The problem is mainly with the way that you're generating tests examples which is resulting into overflow. Kindly refer to the tests of the numpy fronted function here
> 
> https://github.com/unifyai/ivy/blob/59e6f8fd57978501724dcb6425709171e1368584/ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_sums_products_differences.py#L77
> 
> And do check the comments below. Thanks

hello @hirwa-nshuti thank you for the reply. 
this is the error i am getting while locally running test

TypeError: Cannot cast array data from dtype('uint8') to dtype('bool') according to the rule 'safe'

so i had to explicitly convert 'where' array to 'bool'.
I have referred backend function and i have done the same implementation in frontend. still getting these errors.
Thank you

**Comment by AkshayC2001** [comment]
hello @hirwa-nshuti 
it seems like my solution worked and all tests are passing.
please can you review the code? 
Thank you

**Comment by hirwa-nshuti** [comment]
> hello @hirwa-nshuti it seems like my solution worked and all tests are passing. please can you review the code? Thank you

The problem is that your solution might work but it will break other things while the test of this method is possible simple to modify following the example I shared with you.

**Review by hirwa-nshuti — APPROVED** [review]
Looks good now!

Going to merge😊
Thanks for contributing

**Pull request merged**